tf-ncl/src/intermediate.rs

```diff
@@ -34,7 +34,10 @@ pub enum Type {
         max: Option<u32>,
         content: Box<Type>,
     },
-    Object(HashMap<String, Attribute>),
+    Object {
```

Comment:
Why is this called object? Does this correspond to an entity of Terraform, or just JSON? Then, how it is different from Dictionary below?

Reply:
It's a bit of a mashup between types in Terraform, blocks in Terraform and the representation in Nickel. In some places Terraform works with what it calls "objects", which should correspond pretty directly to Nickel records. In particular, the allowed fields and their types are known.

The `Dictionary` variant is used when a Terraform block allows a label to be arbitrary, e.g.
```hcl
resource google_pubsub_topic "topic_name" {
}
```

This fragment would correspond to something like
```
Object("google_pubsub_topic" => Dictionary{ inner = Object(<block content schema>), .. })
```
schematically.